ipykernel/kernelbase.py

```diff
@@ -290,10 +290,50 @@ def __init__(self, **kwargs):
             self.do_execute, ["cell_meta", "cell_id"]
         )
 
-    def dispatch_control(self, msg):
-        self.control_queue.put_nowait(msg)
+    async def dispatch_control(self, msg):
+        # Ensure only one control message is processed at a time
+        async with asyncio.Lock():
```

Comment:
But we shouldn't use `asyncio` anymore, right?

Reply:
I messed a few things up in the merge conflict effort, I'll have to come back to this.